src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -108,6 +106,42 @@ public PodTemplate getTemplateOrNull() {
         return template;
     }
 
+    /**
+     * Makes a best effort to find the build log corresponding to this agent.
+     */
+    @NonNull
+    public TaskListener getRunListener() {
+        PodTemplate podTemplate = getTemplateOrNull();
+        if (podTemplate != null) {
+            TaskListener listener = podTemplate.getListenerOrNull();
+            if (listener != null) {
+                return listener;
+            }
+        }
+        Computer c = toComputer();
+        if (c != null) {
+            for (Executor executor : c.getExecutors()) {
+                Queue.Executable executable = executor.getCurrentExecutable();
+                // If this executor hosts a PlaceholderExecutable, send to the owning build log.
+                if (executable != null) {
+                    Queue.Executable parentExecutable = executable.getParentExecutable();
+                    if (parentExecutable instanceof FlowExecutionOwner.Executable) {
+                        FlowExecutionOwner flowExecutionOwner = ((FlowExecutionOwner.Executable) parentExecutable).asFlowExecutionOwner();
+                        if (flowExecutionOwner != null) {
+                            try {
+                                return flowExecutionOwner.getListener();
+                            } catch (IOException x) {
+                                LOGGER.log(Level.WARNING, null, x);
+                            }
+                        }
+                    }
+                }
+                // TODO handle freestyle and similar if executable instanceof Run, by capturing a TaskListener from RunListener.onStarted
+            }
```

Comment:
This is exactly the kind of logic I was expecting to be possible, but had no clue how to navigate through the executables so that it would be possible. Really nice!